Userland/Utilities/hexdump.cpp

```diff
@@ -59,20 +67,49 @@ int main(int argc, char** argv)
     };
 
     Array<u8, BUFSIZ> contents;
+    Span<u8> prev_line {};
     static_assert(LINE_LENGTH_BYTES * 2 <= contents.size(), "Buffer is too small?!");
     size_t contents_size = 0;
 
     int nread;
+    auto state = STATES::PRINT;
     while (true) {
         nread = file->read(&contents[contents_size], BUFSIZ - contents_size);
         if (nread <= 0)
             break;
         contents_size += nread;
-        // Print as many complete lines as we can (possibly none).
+
         size_t offset;
         for (offset = 0; offset + LINE_LENGTH_BYTES - 1 < contents_size; offset += LINE_LENGTH_BYTES) {
-            print_line(&contents[offset], LINE_LENGTH_BYTES);
+            if (verbose) {
+                print_line(&contents[offset], LINE_LENGTH_BYTES);
+                continue;
+            }
+
+            auto current_line = Span<u8> { contents.data() + offset, LINE_LENGTH_BYTES };
```

Comment:
Let's use nicer APIs with bounds checking:
```suggestion
            auto current_line = contents.span().slice(offset, LINE_LENGTH_BYTES);
```